### PR TITLE
[13.0][IMP] maintenance_plan: Add Generate preventive maintenance requests button from plan view form.

### PR DESCRIPTION
**Pull request by victoralmau**
Add Generate preventive maintenance requests button from plan view form

Add button from maintenance plan to allow generate maintenance requests without waiting for cron.

Please @pedrobaeza and @CarlosRoca13 can you review it?

@Tecnativa TT41729

### CODE REVIEW

File: maintenance_plan/models/maintenance_plan.py [hunk 1]
```diff
@@ -197,3 +197,9 @@ def unlink(self):
             "equipment maintenance plan.",
         )
     ]
+
+    def action_generate_preventive_maintenance_requests(self):
```
Comment (pedrobaeza) [inline maintenance_plan/models/maintenance_plan.py]:
```suggestion
    def button_manual_request_generation(self):
```

File: maintenance_plan/models/maintenance_plan.py [hunk 2]
```diff
@@ -197,3 +197,9 @@ def unlink(self):
...
+        """Call the cron method that will generate the equipment requests."""
```
Comment (pedrobaeza) [inline maintenance_plan/models/maintenance_plan.py]:
```suggestion
        """Call the same method that the cron for generating manually the maintenance requests."""
```

File: maintenance_plan/views/maintenance_plan_views.xml [hunk 3]
```diff
@@ -15,6 +15,15 @@
         <field name="model">maintenance.plan</field>
         <field name="arch" type="xml">
             <form string="Maintenance Plan">
+                <header>
+                    <button
+                        name="action_generate_preventive_maintenance_requests"
+                        string="Generate preventive maintenance requests"
```
Comment (pedrobaeza) [inline maintenance_plan/views/maintenance_plan_views.xml]:
```suggestion
                        string="Generate requests for current threshold"
                        help="If not clicked, the scheduled action will do it for you."
```

File: maintenance_plan/views/maintenance_plan_views.xml [hunk 4]
```diff
@@ -15,6 +15,15 @@
...
+                        type="object"
+                        class="oe_highlight"
```
Comment (pedrobaeza) [inline maintenance_plan/views/maintenance_plan_views.xml]: No highlight, please. This is not the option to follow.

```suggestion
```

**Review by pedrobaeza — CHANGES_REQUESTED** [review]

**Comment by victoralmau** [comment]
Changes done.

**Comment by victoralmau** [comment]
Rebase done.

**Review by pedrobaeza — APPROVED** [review]

**Review by CarlosRoca13 — APPROVED** [review]
Code review

**Pull request merged**